fastlane/Fastfile

```diff
@@ -97,402 +95,407 @@ lane :test do | options |
+    zip(path: archive_path, output_path: archive_zip_path)
+
+    create_release(
+      repository: GITHUB_REPO,
+      version: ios_get_app_version(public_version_xcconfig_file: VERSION_FILE_PATH),
+      release_notes_file_path: 'Simplenote/Resources/release_notes.txt',
+      release_assets: [archive_zip_path],
+      prerelease: options[:prerelease]
     )
   end
+end
 
-  desc 'Updates the main `Localizable.strings` file — that will be imported by GlotPress'
-  lane :generate_strings_file_for_glotpress do
-    en_lproj_path = File.join(PROJECT_FOLDER, 'Simplenote', 'en.lproj')
-    ios_generate_strings_file_from_code(
-      paths: ['Simplenote', 'Pods/Simperium-OSX/*/SPAuthentication*'],
-      output_dir: en_lproj_path,
-      quiet: true
-    )
+# Upload the localized metadata (from `fastlane/metadata/`) to App Store Connect
+#
+# @option [Boolean] with_screenshots (default: false) If true, will also upload the latest screenshot files to ASC
+#
+desc 'Upload the localized metadata to App Store Connect, optionally including screenshots.'
+lane :update_metadata_on_app_store_connect do |options|
+  # Skip screenshots by default. The naming is "with" to make it clear that
+  # callers need to opt-in to adding screenshots. The naming of the deliver
+  # (upload_to_app_store) parameter, on the other hand, uses the skip verb.
+  with_screenshots = options.fetch(:with_screenshots, false)
+  skip_screenshots = !with_screenshots
+
+  upload_to_app_store(
+    app_identifier: APP_STORE_BUNDLE_IDENTIFIER,
+    app_version: ios_get_app_version(public_version_xcconfig_file: VERSION_FILE_PATH),
+    skip_binary_upload: true,
+    screenshots_path: './screenshots/',
+    skip_screenshots: skip_screenshots,
+    overwrite_screenshots: true, # won't have effect if `skip_screenshots` is true
+    phased_release: true,
+    precheck_include_in_app_purchases: false,
+    api_key_path: APP_STORE_CONNECT_API_KEY_PATH
+  )
+end
+
+# Temporary lane to automate the process of uploading the .app manually
+# exported after archiving the Simplenote scheme from Xcode.
+#
+# The process is not automated via the build_and_upload_beta lane due to
+# issues with Sparkle and code signing.
+lane :upload_app_to_appcenter do |options|
+  app_path = options[:app_path]
+  UI.user_error! "Missing .app path option.\nUsage:\n\tfastlane upload_app_to_appcenter app_path:path/to/.app" if app_path.nil?
+
+  app_path = File.expand_path(app_path)
+  UI.user_error! "Could not find .app at #{app_path}" unless File.exist?(app_path)
 
-    # Convert the UTF-16 encoded strings file to UTF-8.
-    strings_file_path = File.join(en_lproj_path, 'Localizable.strings')
-    utf8_strings = File.read(strings_file_path, encoding: 'UTF-16', mode: 'rb')
-    File.write(strings_file_path, utf8_strings, mode: 'w', encoding: 'UTF-8')
+  Tempfile.create do |t|
+    release_notes_path = File.path(t)
+    system "pushd ..; ruby ./Scripts/extract_release_notes.rb -k > #{release_notes_path}; popd"
```

Comment:
💭  What an odd pattern… using `system` to call `ruby` as a subprocess to run Ruby code… from within a Ruby script / Fastfile… This would benefit from being refactored into a `fastlane/lib/extract_release_notes.rb` file that we can `require` from within the `Fastfile` and then call methods of this ruby file directly from there…

Obviously, not something to fix in this PR, but instead something to keep in mind for future improvements.

Reply:
Agreed! Hopefully this won't be needed anyway when it gets updated to use the current translation tooling.